src/core/transpile.jl

```diff
@@ -1397,7 +1389,7 @@ end
 struct CompressRzGatesTranspiler<:Transpiler end
 
 # construct a PhaseShift gate from an input Operator
-function as_phase_shift_gate(target::Integer,op::AbstractOperator)::PhaseShift
+function as_phase_shift_gate(target::Integer,op::AbstractOperator)::Gate
```

Comment:
could be constrained to Gate{PhaseShift} by modifying Gate struct to be generic

Reply:
Updated